Modules/SubModule_ASL/xASL_wrp_Quantify.m

```diff
@@ -225,19 +239,42 @@ function xASL_wrp_Quantify(x, PWI_Path, OutputPath, M0Path, SliceGradientPath, b
     %% 6.   Initialize quantification parameters
     if ~isfield(x.Q,'nCompartments') || isempty(x.Q.nCompartments) || x.Q.nCompartments>2
         x.Q.nCompartments = 1; % by default, we use a single-compartment model, as proposed by the Alsop et al. MRM 2014 concensus paper
-    end
+	end
 
-    if ~isfield(x.Q,'ATT')
+	if ~isfield(x.Q,'ATT')
         x.Q.ATT = 1800; % ms as default micro-vascular ATT
-    end
-    if ~isfield(x.Q,'TissueT1')
-        x.Q.TissueT1 = 1240; % T1 GM tissue @ 3T
-    end
-    if ~isfield(x.Q,'BloodT1')
-        x.Q.BloodT1  = 1650; % T1 relaxation time of arterial blood DEFAULT=1650 @ 3T
-    end
+	end
+	if ~isfield(x.Q,'TissueT1')
+		switch(x.MagneticFieldStrength)
+			case 3
+				x.Q.TissueT1 = 1240; % T1 GM tissue @ 3T
+			case 1
+				x.Q.TissueT1 = 1000; 
+			case 4
+				x.Q.TissueT1 = 1350;
+			otherwise
+				x.Q.TissueT1 = 1240;
+				fprintf('%s\n',['Warning: Unknown T1 GM for ' num2str(x.MagneticFieldStrength) 'T scanners']);
+		end
+	end
+	if ~isfield(x.Q,'BloodT1')
+		switch(x.MagneticFieldStrength)
+			case 3
+				x.Q.BloodT1 = 1650; % T1 relaxation time of arterial blood DEFAULT=1650 @ 3T
+			case 1.5
+				x.Q.BloodT1 = 1350;
+			case 7
+				x.Q.BloodT1 = 2100;
+			otherwise
+				fprintf('%s\n',['Warning: Unknown T1-blood for ' num2str(x.MagneticFieldStrength) 'T scanners']);
+		end
+	end
     if ~isfield(x.Q,'T2art')
-        x.Q.T2art = 50; % T2* of arterial blood, only used when no M0 image
+		if x.MagneticFieldStrength == 3
+			x.Q.T2art = 50; % T2* of arterial blood, only used when no M0 image
+		else
+			fprintf('%s\n','Warning: Unknown T2-art for non-3T scanners');
```

Comment:
We can consider only issuing this warning when needed. `x.Q.T2art` is only used with an M0 value, or when the T2 differs between deltaM and M0.

Reply:
This one is never used, so I have remove the warning.